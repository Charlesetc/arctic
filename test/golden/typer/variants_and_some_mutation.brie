define main [
        define x < field = (One 2) >
        while true [
                if true [
                        x.field = Two "hi"                                             
                ] else [
                        x.field = One 3
                ]
        ]
]
