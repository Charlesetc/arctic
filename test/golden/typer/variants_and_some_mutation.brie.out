root[(ident(define):: ident(main):: block[][(ident(define):: ident(x):: <field = ((ident(One):: ident(2)::Integer)::[^ One Integer ])::[^ One Integer ]>::<field = [^ One Integer, Two String ]>)::, (ident(while):: (ident(true)::Bool)::Bool block[][(ident(if):: (ident(true)::Bool)::Bool block[][(ident(_update):: (ident(x)::<field = [^ One Integer, Two String ]>.field::[^ One Integer, Two String ])::[^ One Integer ] (ident(Two):: string(hi)::String)::[^ Two String ])::]:: block[][(ident(_update):: (ident(x)::<field = [^ One Integer, Two String ]>.field::[^ One Integer, Two String ])::[^ One Integer, Two String ] (ident(One):: ident(3)::Integer)::[^ One Integer ])::]::)::]::)::Unit]::Function)::]::